### ISSUE TITLE
KeePassXC reports the database is locked by another instance when disk space is full

### ISSUE THREAD

**Issue by marc2377** [keepassxreboot/keepassxc#1061]
When attempting to open a password database from a filesystem with zero free space, KeePassXC will report:

> The database you are trying to open is locked by another instance of KeePassXC.
> 
> Do you want to open it anyway?

With options `Open read-only`, `No` and `Yes`.

This message is misleading, as there is no other instance of KeePassXC open.

The program should probably detect that the filesystem is full, and report it to the user.

FWIW this is also a bug in KeePassX.

## Debug Info

KeePassXC - Version 2.2.0
Revision: caa49a8ef3ee28ed478192389b21d61107b3b8e0

Libraries:
- Qt 5.9.2
- libgcrypt 1.8.1

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 4.13.5-1-ARCH

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey

**Comment by droidmonkey** [comment]
Does this file system have zero free space or is it read only (or both)?

**Comment by phoerious** [comment]
The message really shouldn't appear unless you have a stale lock file somewhere. Could you please check?

**Comment by marc2377** [comment]
The file system has zero free space, but it is **not** read-only. If I simply delete or move something from it, and try to open the database again, KeePassXC no longer reports the message.

I'm positive this is not caused by a lock file.

Here is how to reproduce the issue:

```
dd if=/dev/zero of=tmpfs bs=1M count=10
mkfs.ext4 tmpfs
mkdir tmpmount
sudo mount tmpfs tmpmount
sudo chown [user] tmpmount/
```

At this point:
- Launch KeePassXC
- Create a new password database and store it in the newly-created folder `tmpmount`
- Create a password entry, save it and close KeePassXC
- Launch KeePassXC and make sure you can open the newly created database file
- Quit KeePassXC once more.

Then

`dd if=/dev/zero of=./tmpfs/fill bs=1M count=10`

Now launch KeePassXC again and try to open the database file.

Thanks and let me know if more help is needed.

**Comment by droidmonkey** [comment]
it is probably because the lock file cannot be created. The error is misleading.

**Comment by droidmonkey** [comment]
Will be corrected through #1002, tracking all lock file issues there.
